### PR TITLE
the interpreter ignored dicts to be displayed as whole and ignored th…

### PR DESCRIPTION
**Pull request by eyalsn-stratoscale**
…e fix point format for a floating point number

### CODE REVIEW

**Comment by eyalsn-stratoscale** [comment]
I also added support for named groups in regex using XRegExp

**Comment by eyalsn-stratoscale** [comment]
@eyala-stratoscale thanks for review

**Comment by eyala-stratoscale** [comment]
looks fine, @guykr-stratoscale please have a look as well

**Comment by motty-stratoscale** [comment]
Looks great.

File: src/interpreter.js [hunk 1]
```diff
@@ -147,10 +147,16 @@ function created(utcSeconds) {
   return ['<span class="created">', utcSeconds.toFixed(6), ' </span><span class="date">', dateString, ' </span>'].join("")
 }
 
-function argument(arg) {
+function argument(arg, fractional) {
   if (arg == null) {
     return "(null)"
   }
+  if (typeof(arg) == "number" && typeof(fractional) != "undefined") {
+    var fix = parseInt(fractional)
```
Comment (guykr) [inline src/interpreter.js]: `parseInt(fractional, 10)`

File: src/XRegExp.js [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+        return regex.exec(this); // Run the altered `exec`
+    };
+
+    // Adds support for `${n}` tokens for named and numbered backreferences in replacement text,
+    // and provides named backreferences to replacement functions as `arguments[0].name`. Also
+    // fixes cross-browser differences in replacement text syntax when performing a replacement
+    // using a nonregex search value, and the value of replacement regexes' `lastIndex` property
+    // during replacement iterations. Note that this doesn't support SpiderMonkey's proprietary
+    // third (`flags`) parameter
+    String.prototype.replace = function (search, replacement) {
```
Comment (guykr) [inline src/XRegExp.js:334]: wow, this is really bad practice
let's hope this library doesn't have bugs...

**Comment by guykr** [comment]
shipit

**Pull request merged**